src/sknnr_spatial/image/_dask_backed.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import dask.array as da
+from sklearn.utils.validation import check_is_fitted
+
+if TYPE_CHECKING:
+    import pandas as pd
+    from numpy.typing import NDArray
+    from sklearn.base import BaseEstimator
+    from sklearn.neighbors import KNeighborsRegressor
+
+    from ..types import DaskBackedType
+    from .dataarray import DataArrayPreprocessor
+    from .dataset import DatasetPreprocessor
+
+
+class TargetInferenceError(Exception):
+    """Raised when the number of targets cannot be inferred from an estimator."""
+
+
+def predict_from_dask_backed_array(
+    X_image: DaskBackedType,
+    *,
+    estimator: BaseEstimator,
+    y=None,
+    preprocessor_cls: type[DataArrayPreprocessor] | type[DatasetPreprocessor],
+    nodata_vals=None,
+) -> DaskBackedType:
+    """Generic predict wrapper for Dask-backed arrays."""
+    check_is_fitted(estimator)
+    preprocessor = preprocessor_cls(X_image, nodata_vals=nodata_vals)
+
+    try:
+        n_targets = _infer_num_targets(y, estimator)
+        target_names = _infer_target_names(y, n_targets)
```

Comment:
Another question here.  It looks like `_infer_target_names` won't raise an exception.  Does it make sense to keep it in this `try` block?  Probably no harm if it remains here, though.

Reply:
You're right, better practice to keep the try block small!